quickwit-index-config/src/default_index_config/default_config.rs

```diff
@@ -214,11 +253,20 @@ impl IndexConfig for DefaultIndexConfig {
             DocParsingError::NotJson(doc_json_sample)
         })?;
         let parsing_result = self.field_mappings.parse(&json_obj)?;
+        let tags_field_opt = self.schema.get_field(TAGS_FIELD_NAME);
         for (field_path, field_value) in parsing_result {
+            let field_name = field_path.field_name();
             let field = self
                 .schema
-                .get_field(&field_path.field_name())
-                .ok_or_else(|| DocParsingError::NoSuchFieldInSchema(field_path.field_name()))?;
+                .get_field(&field_name)
+                .ok_or_else(|| DocParsingError::NoSuchFieldInSchema(field_name.clone()))?;
+            if self.tag_field_names.contains(&field_name) {
+                let tags_field = tags_field_opt.ok_or_else(|| {
+                    DocParsingError::NoSuchFieldInSchema(TAGS_FIELD_NAME.to_string())
```

Comment:
I would raise the exception immediately when doing `self.schema.get_field(TAGS_FIELD_NAME)``

Reply:
Hmm, I preferred the option so that the calling client decides what Error to raise since we tend to do field resolution in two places.

Reply:
Not sure to get it :). Anyway this is very minor.